sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -614,7 +614,11 @@ object ViewHelper extends SQLConfHelper with Logging {
     }.getOrElse(false)
     if (replace && uncache) {
       logDebug(s"Try to uncache ${name.quotedString} before replacing.")
-      checkCyclicViewReference(analyzedPlan, Seq(name), name)
+      if (!conf.storeAnalyzedPlanForView) {
```

Comment:
how about `!conf.storeAnalyzedPlanForView && originalText.nonEmpty`? It also matches the condition below where we decide to store analyzed plan.

Reply:
Sure, I just want to be conservative in the first version. Because `!conf.storeAnalyzedPlanForView && originalText.nonEmpty` means dataset view will always skip cyclic check. It's consistent with 3.1 which is good. But I thought we should always block recursive views unless the user explicitly changes the flag (even if the recursive views work well).